USF-1136 Docs for creating experiments (#154)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/all/experimenting.mdx

description: Learn about how to create A/B experiments for drop-in components.
---

import { Tabs, TabItem } from '@astrojs/starlight/components';
import Diagram from '@components/Diagram.astro';
import Vocabulary from '@components/Vocabulary.astro';
import Aside from '@components/Aside.astro';
import Callouts from '@components/Callouts.astro';
import { Steps } from '@astrojs/starlight/components';
import Tasks from '@components/Tasks.astro';
import Task from '@components/Task.astro';

This topic introduces you to the concept of creating A/B experiments for drop-in components. You will learn how to create experiments, define variants, and track the performance of each variant.

## Vocabulary

<Vocabulary>

### A/B experiment

A test that compares two or more pages to determine which page performs better.

### Variant

A version of a page that is different from the original version. Variants are created to test different features, designs, or content.

### Control

The original version of the drop-in component that is used as a baseline for comparison.

### Challenger

The variant of the drop-in component being tested.

</Vocabulary>

## Step-by-step

> The steps for creating an A/B experiment for a commerce storefront page, are summarized below, and detailed in the following tasks:

Reviewer: ```suggestion
Creating an A/B experiment for a commerce storefront page includes the following tasks:
```

<Steps>
1. Create a variant page for the experiment.
1. Add the experiment to the metadata block of the original page.

> 1. Preview the experiment using AEM Sidekick Preview button.

Reviewer: ```suggestion
1. Preview the experiment using the AEM Sidekick extension.
```

1. Make code changes for experiments, if necessary.
</Steps>

<Tasks>
<Task>

> ### Create an variant page for the experiment

Reviewer: ```suggestion
### Create a variant page for the experiment
```

> Duplicate the original page and make the necessary changes to create a new variant page for the experiment. Update any placeholder labels or content in the variant page to reflect the changes you want to test. For example, you can change the color of a button, update the text of a call-to-action, or rearrange the layout of the page. Save the variant page with a new name to distinguish it from the original page.

Reviewer: Consider linking to the page about labels or adding "labels" to the vocabulary section since I think they're unique to  drop-in components and that might not be clear to the uninitiated.

</Task>

<Task>
### Add the experiment to the original page

> Add the experiment to the metadata block of the original page by adding the experiment name and the URL to the experiment page. For example:

Reviewer: I'm assuming that this is intentionally vague, but it would be good to follow up with some specific examples of experimentation. There was at least one example in the GA demo, though I don't recall specifically what it was.

Author: On it!

| Metadata                      | |
| ------------------------ | -------------------------- |
| Title     | Cart      |
| Robots | noindex,nofollow |
| Experiment | CART001 |

> | Experiment Variants | https://marketer-demo---aem-boilerplate-commerce--hlxsites.aem.page/drafts/marketer/cart-new |

Reviewer: Is this meant to resolve? If not, let's wrap it in backticks to prevent rendering as a link.

</Task>

<Task>
### Preview the experiment

Use the AEM Sidekick Preview button to preview the experiment and compare the original page with the variant page. Make sure that the changes are displayed correctly and that the experiment is functioning as expected. You can also share the experiment URL with stakeholders for feedback before launching it to a wider audience.

<Diagram caption="Preview the experiment">![Preview the experiment](@images/ExperimentOne.png)</Diagram>

</Task>

<Task>
### Code changes for experiments

> Some experiments, such as creating variants of labels with the `placeholders` file, require code changes to a drop-in's initializer code. For example, by making the following additions to the default `cart.js` initializer, you can read the metadata block from the cart page and identify when a variant is being delivered, then apply an alternate `placeholders` file.

Reviewer: ```suggestion
Some experiments, such as creating variants of labels with the `placeholders` file, require code changes to a drop-in component's initializer code. For example, by making the following additions to the default `cart.js` initializer, you can read the metadata block from the cart page and identify when a variant is being delivered, then apply an alternate `placeholders` file.
```
